### PR TITLE
 Add SonarCloud

### LINKED ISSUE #14: Add Sonarcloud and jacoco

### PR DESCRIPTION
**Pull request by Lucia560**
I removed ci/cd pipeline as it had the wrong format, I added the sonar.yml and  maven.yml (correct pipeline). Also I realised that I forgot to add  jacoco so this will be  a draft .

### CODE REVIEW

**Comment by TrungMa1** [comment]
Could we implement the function that delete branches after successfull merge? Like we did in our first laboration?
Looking back at our Jakarta labb i think it was putting this into maven.yml that added that.

      - name: Cleanup merged branches
        if: github.event_name == 'pull_request' && github.event.action == 'closed' && github.event.pull_request.merged == true
        run: |
          git fetch --prune
          git push origin --delete "${{ github.event.pull_request.head.ref }}"

**Comment by Lucia560** [comment]
I did that via github, everytime we merge a branch is deleted. Now the fails I get here are due to the tests and the missing enviroments for the tests. I will let is like this for now, and come back to it after I make a few tests.

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+
```
Comment (TrungMa1) [inline .github/workflows/maven.yml:14]: If you delete everything from row 15-32, should'nt this fix the #24

**Comment by Lucia560** [comment]
Yes. I will fix it in another PR together with the flyway.

**Pull request closed without merging**